admin-api-server/src/nft/service/file/file.service.spec.ts

```diff
@@ -0,0 +1,357 @@
+
+          const thumbnailDimensions = sizeOf(thumbnail!.buffer);
+          expect(thumbnailDimensions.width).toBe(250);
+          expect(thumbnailDimensions.height).toBeLessThan(250);
+
+          const afterManipulation = res.filter(
+            (file) =>
+              file.originalname === 'artifact' ||
+              file.originalname === 'display',
+          );
+          expect(afterManipulation).toStrictEqual([artifact, display]);
+        });
+      });
+
+      describe('when only display is missing', () => {
+        it('display should be created with size 350x350', async () => {
+          const buffer = fs.readFileSync(__dirname + '/shortvideo.spec.mov');
+          const artifact = artifactMock(buffer, 'video');
+          const thumbnail = await thumbnailMock();
+          const filesArray = [artifact, thumbnail];
+
+          const res = await fileService.addMissingFiles(filesArray);
+          const display = res.find((file) => file.originalname === 'display');
+
+          expect(res.length).toBe(3);
+          expect(isImage(display!)).toBe(true);
+
+          expect(expectedDimensions(display!, 350, 350)).toBe(true);
+
+          const afterManipulation = res.filter(
+            (file) =>
+              file.originalname === 'artifact' ||
+              file.originalname === 'thumbnail',
+          );
+          expect(afterManipulation).toStrictEqual([artifact, thumbnail]);
+        });
+      });
+    });
+
+    describe('if the artifact is an image', () => {
+      describe('when display and thumbnail are missing', () => {
+        it('and the artifact has a smaller size than 350, display and thumbnail should be copies of the artifact', async () => {
+          const smallImageBuffer = fs.readFileSync(
+            __dirname + '/smallimage.spec.jpeg',
+          );
+          const buffer = await sharp(smallImageBuffer)
+            .resize({ width: 300, height: 300 })
+            .toBuffer();
+          const artifact = artifactMock(buffer, 'image');
+          const filesArray = [artifact];
+
+          const res = await fileService.addMissingFiles(filesArray);
+          const display = res.find((file) => file.originalname === 'display');
+          const thumbnail = res.find(
+            (file) => file.originalname === 'thumbnail',
+          );
+
+          expect(res.length).toBe(3);
+          expect(isImage(display!)).toBe(true);
+          expect(isImage(thumbnail!)).toBe(true);
+
+          expect(expectedDimensions(display!, 300, 300)).toBe(true);
+          expect(expectedDimensions(thumbnail!, 300, 300)).toBe(true);
+
+          const artifactAfterManipulation = res.find(
+            (file) => file.originalname === 'artifact',
+          );
+          expect(artifactAfterManipulation).toStrictEqual(artifact);
+        });
+        it('and the artifact has a larger size than 350, display should be a scaled version with size 350 and thumbnail with size 250', async () => {
```

Comment:
Display shouldn't target an artifically lower resolution. It should simply be the same resolution as what can be directly obtained from the artifact (so if it's a movie, with lets say 780x1200 resolution, then the display should be an image from within that movie of also 780x1200 resolution).

If the artifact is an image, then the display should just point to the same asset.